src/switch.py

```diff
@@ -0,0 +1,275 @@
+#!/usr/bin/env python3
+#
+# igcollect - Switch stats:
+#             - port traffic
+#             - port errors
+#             - cpu usage
+#
+# Copyright (c) 2017, InnoGames GmbH
+#
+
+from argparse import ArgumentParser
+from time import time
+from pysnmp import proto, error
+from pysnmp.hlapi import (
+        CommunityData,
+        ContextData,
+        getCmd,
+        bulkCmd,
+        ObjectType,
+        ObjectIdentity,
+        SnmpEngine,
+        UdpTransportTarget,
+        UsmUserData,
+        usmHMACSHAAuthProtocol,
+        usmDESPrivProtocol,
+    )
+
+import re
+import sys
+
+snmp_engine = SnmpEngine()
+context_data = ContextData()
+
+
+OIDs = {
+    'switch_model': '.1.3.6.1.2.1.1.1.0',
+    'port_name': '.1.3.6.1.2.1.31.1.1.1.1',
+    'port_state': '1.3.6.1.2.1.2.2.1.8',
+}
+
+LAGG_OIDs = {
+    'procurve': '.1.3.6.1.4.1.11.2.14.11.5.1.7.1.3.1.1.8',
+    'powerconnect': '.1.2.840.10006.300.43.1.2.1.1.12',
+}
+
+CPU_OIDs = {
+    'procurve': '.1.3.6.1.4.1.11.2.14.11.5.1.9.6.1.0',
+    'powerconnect': '.1.3.6.1.4.1.674.10895.5000.2.6132.1.1.1.1.4.4.0',
+    'extreme': '.1.3.6.1.4.1.1916.1.32.1.2.0',
+}
+
+COUNTERS = {
+    'bytesIn': '.1.3.6.1.2.1.31.1.1.1.6',
+    'bytesOut': '.1.3.6.1.2.1.31.1.1.1.10',
+    'pktsIn': '.1.3.6.1.2.1.31.1.1.1.7',
+    'pktsOut': '.1.3.6.1.2.1.31.1.1.1.11',
+    'brdPktsIn': '.1.3.6.1.2.1.31.1.1.1.9',
+    'brdPktsOut': '.1.3.6.1.2.1.31.1.1.1.13',
+    'ifInErrors': '.1.3.6.1.2.1.2.2.1.14',
+    'ifOutErrors': '.1.3.6.1.2.1.2.2.1.20',
+    'ifInDiscards': '.1.3.6.1.2.1.2.2.1.13',
+    'ifOutDiscards': '.1.3.6.1.2.1.2.2.1.19',
+}
+
+
+class SwitchException(Exception):
+    pass
+
+
+def main():
+    args = parse_args()
+    if not args.prefix:
+        args.prefix = 'switches.{}'.format(args.switch)
+    try:
+        snmp = get_snmp_connection(args)
+    except error.PySnmpError as e:
+        print(e, file=sys.stderr)
+        return -1
+
+    try:
+        model = get_switch_model(snmp)
+    except SwitchException as e:
+        print(e, file=sys.stderr)
+        return -1
+
+    if not model:
+        return -1
+
+    monitored_ports = get_monitored_ports(snmp, model)
+    ports_stats(args.prefix, snmp, monitored_ports)
+    cpu_stats(args.prefix, snmp, model)
+
+
+def parse_args():
+    parser = ArgumentParser()
+    parser.add_argument('switch', type=str, help='Hostname of a switch')
+    parser.add_argument('--prefix', help='Graphite prefix')
```

Comment:
I haven't noticed that you are assigning a default later.  I guess it is okay than.  All other scripts simply setting their name as the default prefix.  `nginx.py` would print metrics like `nginx.xxx`.  We could do something more complicated by getting the hostname, but I preferred not to do so, as this would most likely be different for everybody.